File: SW Expert Academy/D1/2056. 연월일 달력.cpp
/*******************************************
 * SW Expert Academy 2056. 연월일 달력
 * D1
 * 21.2.2 try: 2
 *******************************************/
#include <iostream>
#include <string>

int main(int argc, char** argv) {
	std::ios_base::sync_with_stdio(false);
	std::cin.tie(nullptr);
	std::cout.tie(nullptr);

	int test_case;
	int T;
	
	//freopen("input.txt", "r", stdin); //#include <cstdio>
	std::cin >> T;

	for (test_case = 1; test_case <= T; ++test_case) {
		std::string input;
		std::string ans = "-1";

		std::cin >> input;

		std::string year = input.substr(0, 4);
		std::string month = input.substr(4, 2);
		std::string day = input.substr(6, 2);

		switch (std::stoi(month)) {
		case 1:
		case 3:
		case 5:
		case 7:
		case 8:
		case 10:
		case 12:
			if (day > "0" && day <= "31")
				ans = year + '/' + month + '/' + day;
			break;
		case 4:
		case 6:
		case 9:
		case 11:
			if (day > "0" && day <= "30")
				ans = year + '/' + month + '/' + day;
			break;
		case 2:
			if (day > "0" && day <= "28")
				ans = year + '/' + month + '/' + day;
			break;
		default:
			break;
		}
		
		std::cout << "#" << test_case << ' ' << ans << '\n';
	}
	
	return 0;
}

File: SW Expert Academy/D1/2056. 연월일 달력.py
'''
 # SW Expert Academy 2056. 연월일 달력
 # D1
 # 21.2.2 try: 1
'''
#sys.stdin = open("input.txt", "r") #import sys

T = int(input())
for test_case in range(1, T + 1):
    # l = [int(x) for x in input().split()]
    inputs = input()
    ans = "-1"

    year = inputs[0:4]
    month = inputs[4:6]
    day = inputs[6:]

    if (month == "01" or month == "03" or month == "05" or month == "07" or month == "08" or month == "10" or month == "12") and "0" < day <= "31":
        ans = year + '/' + month + '/' + day;
    elif (month == "04" or month == "06" or month == "09" or month == "11") and "0" < day <= "30":
        ans = year + '/' + month + '/' + day;
    elif month == "02" and "0" < day <= "28":
        ans = year + '/' + month + '/' + day;


    print(f"#{test_case} {ans}")